### DISCUSSION TITLE
3-terrarium/3-intro-to-DOM-and-closures/README

### DISCUSSION THREAD

**Discussion by ultrasounder** [microsoft/Web-Dev-For-Beginners#556] [General]
Hi working through this and found a bug. The `function displayCandy(){
    let candy = ['jellybeans'];
    function addCandy(candyType) {
        candy.push(candyType)
    }
    addCandy('gumdrops');
}
displayCandy();
console.log(candy)`
throws an error as the displayCandy function expects the console.log statement to be inside the function body. It should be like this;
`function displayCandy(){
    let candy = ['jellybeans'];
    function addCandy(candyType) {
        candy.push(candyType)
    }
    addCandy('gumdrops');
  console.log(candy)
}
displayCandy();`
I ran this in Codepen and confirmed the result. Would be glad to submit a PR.

**Comment by ultrasounder** [comment]
![image](https://user-images.githubusercontent.com/56858473/146711861-af0c7b12-aeb7-442a-bd3c-e8a7b5fb6c9a.png)

**Comment by jlooper** [comment]
hi, if you read the lesson text, indeed, an error will be thrown. The lesson is about how to avoid this type of problem by using closures.

**Reply by ultrasounder** [reply]
thanks @jlooper totally enjoying the course BTW!
